hydra_base/lib/scenario.py

```diff
@@ -1377,9 +1411,15 @@ def get_resource_data(ref_key,
 
     user_id = kwargs.get('user_id')
 
-    resource_i = get_resource(ref_key, ref_id)
-
-    resource_attributes = resource_i.attributes
+    if ref_key is not None and ref_id is not None:
+        resource_i = get_resource(ref_key, ref_id)
+        resource_attributes = resource_i.attributes
+    elif ref_key is None and ref_id is None:
+        scenario = _get_scenario(scenario_id, user_id)
+        resource_attributes = _get_all_network_resource_attributes(scenario.network_id)
+    elif None in (ref_key, ref_id): # Both are None
```

Comment:
Comment for `1420` should read "One is None" as `1417` has taken the case where both are None.

So the intention here is to return all attrs from the scenario when neither `ref_key` nor `ref_id` are provided, but raise if only one of the two is present?

Reply:
Doh. Fixed. Good spot.